contrib/dracut/90zfs/zfs-load-key.sh.in

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+
+# This script only gets executed on systemd systems, see mount-zfs.sh for non-systemd systems
+
+# There is a race between the zpool import and the pre-mount hooks, so we
+#   we wait for a pool to be imported
+while true; do
+    if $(zpool list -H | grep -q -v '^$'); then
```

Comment:
No dollar parentheses here.  You're grepping for its retval, not trying to obtain the contents of the grep.